warehouse/schema/schema_v2.go

```diff
@@ -0,0 +1,216 @@
+package schema
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/utils/timeutil"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/repo"
+	whutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+type schemaV2 struct {
+	stats struct {
+		schemaSize stats.Histogram
+	}
+	warehouse                        model.Warehouse
+	log                              logger.Logger
+	ttlInMinutes                     time.Duration
+	schemaRepo                       schemaRepo
+	stagingFilesSchemaPaginationSize int
+	stagingFileRepo                  stagingFileRepo
+	enableIDResolution               bool
+	fetchSchemaRepo                  fetchSchemaRepo
+	now                              func() time.Time
+	cachedSchema                     model.Schema
+	cacheExpiry                      time.Time
+	cachedSchemaMu                   sync.RWMutex
+}
+
+func newSchemaV2(v1 *schema, warehouse model.Warehouse, log logger.Logger, schemaSize stats.Histogram, ttlInMinutes time.Duration, fetchSchemaRepo fetchSchemaRepo) *schemaV2 {
+	v2 := &schemaV2{
+		warehouse:                        warehouse,
+		log:                              log,
+		ttlInMinutes:                     ttlInMinutes,
+		schemaRepo:                       v1.schemaRepo,
+		stagingFilesSchemaPaginationSize: v1.stagingFilesSchemaPaginationSize,
+		stagingFileRepo:                  v1.stagingFileRepo,
+		fetchSchemaRepo:                  fetchSchemaRepo,
+		enableIDResolution:               v1.enableIDResolution,
+		now:                              timeutil.Now,
+	}
+	v2.stats.schemaSize = schemaSize
+	return v2
+}
```

Comment:
Since we are already passing v1, can we directly use that instead of passing schemaSize?
```suggestion
func newSchemaV2(v1 *schema, warehouse model.Warehouse, log logger.Logger, ttlInMinutes time.Duration, fetchSchemaRepo fetchSchemaRepo) *schemaV2 {
	v2 := &schemaV2{
		warehouse:                        warehouse,
		log:                              log,
		ttlInMinutes:                     ttlInMinutes,
		schemaRepo:                       v1.schemaRepo,
		stagingFilesSchemaPaginationSize: v1.stagingFilesSchemaPaginationSize,
		stagingFileRepo:                  v1.stagingFileRepo,
		fetchSchemaRepo:                  fetchSchemaRepo,
		enableIDResolution:               v1.enableIDResolution,
		now:                              timeutil.Now,
	}
	v2.stats.schemaSize = v1.stats.schemaSize
	return v2
}
```

Reply:
Done